module/VuFind/src/VuFind/Content/Covers/RecordData.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * RecordData cover content loader.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Universitätsbibliothek Mannheim 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Stefan Weil <sw@weilnetz.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\Content\Covers;
+
+use VuFind\Record\Loader as RecordLoader;
+
+/**
+ * RecordData cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Stefan Weil <sw@weilnetz.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class RecordData extends \VuFind\Content\AbstractCover
+{
+    protected RecordLoader $recordLoader;
+
+    /**
+     * Constructor
+     *
+     * @param RecordLoader $recordLoader Record loader
+     */
+    public function __construct(RecordLoader $recordLoader)
+    {
+        $this->recordLoader = $recordLoader;
```

Comment:
You can use constructor property promotion to simplify this and avoid the need to add more PHPdoc to the property:

```suggestion
    /**
     * Constructor
     *
     * @param RecordLoader $recordLoader Record loader
     */
    public function __construct(protected RecordLoader $recordLoader)
    {
```